### PR TITLE
Add the task unit test in k8s registration API

### PR DESCRIPTION
**Pull request by johnliu25**
Add the task unit test in k8s registration API

### CODE REVIEW

File: karbon/karbon_cluster_registration_service_test.go [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
 	"fmt"
 	"net/http"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/keploy/go-sdk/integrations/khttpclient"
 	"github.com/keploy/go-sdk/keploy"
 	"github.com/keploy/go-sdk/mock"
+	v3 "github.com/nutanix-cloud-native/prism-go-client/v3"
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_service_test.go:14]: the imports from the same repo should be in the thrid group

File: karbon/karbon_cluster_registration_service_test.go [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
 	"github.com/nutanix-cloud-native/prism-go-client/internal/testhelpers"
 )
 
 const (
-	noCategoryMappingErrorMsg = "No category mappings provided"
-	noUUIDErrorMsg            = "uuid in body is required"
+	noCategoryMappingErrorMsg  = "No category mappings provided"
+	noUUIDErrorMsg             = "uuid in body is required"
+	noTaskIDCreateRespErrorMsg = "Task ID was not include the create response."
+	timeout                    = 60 * time.Second
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_service_test.go:26]: consider elaborating on the timeout name e.g. `testWaitTimeout`

File: karbon/karbon_cluster_registration_service_test.go [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
+	taskSucceed                = "SUCCEEDED"
+	taskFailed                 = "FAILED"
+	taskAborted                = "ABORTED"
+	taskSuspended              = "SUSPENDED"
```
Comment (thunderboltsid) [inline karbon/karbon_cluster_registration_service_test.go:30]: Consider prefixing unexported globals with `_` for readability. That way we know it's a global without having to jump to the declaration.
https://github.com/uber-go/guide/blob/master/style.md#prefix-unexported-globals-with-_

Comment (johnliu25) [reply karbon/karbon_cluster_registration_service_test.go:30]: thanks for the recommendation

**Review by thunderboltsid — APPROVED** [review]

File: karbon/karbon_cluster_registration_service_test.go [hunk 4]
```diff
@@ -95,6 +105,28 @@ func validateK8sClusterRegistrationList(t *testing.T, clusterRegList *K8sCluster
...
+		if retryTimes > int(maxRetryTimes) {
+			require.FailNow(t, fmt.Sprintf("Task %s was failed: Task was not completed by timeout.\n", taskID))
+			break
+		}
+		retryTimes += 1
+		responseGetTask, err := v3Client.V3.GetTask(kctx, taskID)
+		require.NoError(t, err)
+		require.NotNil(t, responseGetTask.Status)
+		taskStatus := *responseGetTask.Status
+		if taskStatus == _taskSucceed {
```
Comment (deepakm-ntnx) [inline karbon/karbon_cluster_registration_service_test.go:122]: what if the task is still running forever. should we not fail as well

**Review by deepakm-ntnx — APPROVED** [review]

**Pull request merged**